tools/cooja/apps/mspsim/build.gradle

```diff
@@ -0,0 +1,61 @@
+apply plugin: "java"
+
+sourceSets {
+    main {
+        java {
+            srcDir 'src'
+        }
+    }
+}
+
+task buildMspsimJar(type: GradleBuild) {
+    buildFile = 'importedmspsimBuild.gradle'
+    tasks = ['jar']
+}
+task buildCoffeeJar(type: GradleBuild) {
+    buildFile = '../../../coffee-manager/build.gradle'
+    tasks = ["jar"]
+}
+task copyMspsimJar(type: Copy, dependsOn: buildMspsimJar) {
+    from "../../../mspsim/mspsim.jar"
+    into 'lib'
+}
+task copyJipv6Jar(type: Copy) {
+    from '../../../mspsim/lib/jipv6.jar'
+    into 'lib'
+}
+task copyCoffeeJar(type: Copy, dependsOn: buildCoffeeJar) {
+    from '../../../coffee-manager/coffee.jar'
+    into 'lib'
+}
+compileJava.dependsOn copyMspsimJar
+compileJava.dependsOn copyCoffeeJar
+compileJava.dependsOn copyJipv6Jar
+
+repositories {
+   jcenter()
+   flatDir {
+       dirs 'lib'
+   }
+}
+
+project.gradle.buildFinished { buildResult ->
+  if (buildResult.getFailure() != null) {
+    println "Could not find the MSPSim build file. Did you run &quot;git submodule update --init&quot;?"
```

Comment:
I like the Did you run. It sparks curiosity, which I think is a good thing
;-)

2015-09-29 9:52 GMT+02:00 Moritz Strübe <notifications@github.com>:

> In tools/cooja/apps/mspsim/build.gradle
> <https://github.com/contiki-os/contiki/pull/771#discussion_r40643185>:
>
> > +    into 'lib'
> > +}
> > +compileJava.dependsOn copyMspsimJar
> > +compileJava.dependsOn copyCoffeeJar
> > +compileJava.dependsOn copyJipv6Jar
> > +
> > +repositories {
> > +   jcenter()
> > +   flatDir {
> > +       dirs 'lib'
> > +   }
> > +}
> > +
> > +project.gradle.buildFinished { buildResult ->
> > +  if (buildResult.getFailure() != null) {
> > +    println "Could not find the MSPSim build file. Did you run &quot;git submodule update --init&quot;?"
>
> As it seems that we expect too much from some people, maybe we should
> re-phase it to: Please run: git submodule update --init - though by now
> all those people automatically end up on my ignore-list. ;)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/contiki-os/contiki/pull/771/files#r40643185>.
>